.github/workflows/cli-e2e-tests.yaml

```diff
@@ -36,29 +36,40 @@ jobs:
           AWS_DOCKER_ARTIFACT_REPO: ${{ secrets.AWS_DOCKER_ARTIFACT_REPO }}
           GIT_CREDENTIALS: ${{ secrets.GIT_CREDENTIALS }}
 
+      # Install Poetry.
+      - uses: snok/install-poetry@v1
+        with:
+          version: 1.2.2
+
+      # Install the dependencies for the CLI testing framework.
+      - name: Install CLI testing framework dependencies
+        run: |
+          cd ./crates/aptos/e2e
+          poetry install
+
       # Run CLI tests against local testnet built from devnet branch.
       - uses: nick-fields/retry@7f8f3d9f0f62fe5925341be21c2e8314fd4f7c7c # pin@v2
         name: devnet-tests
         with:
           max_attempts: 5
           timeout_minutes: 20
-          command: cd ./crates/aptos/e2e && python main.py -d --base-network devnet --image-repo-with-project ${{ secrets.GCP_DOCKER_ARTIFACT_REPO }} --test-cli-tag ${{ inputs.GIT_SHA }} --working-directory ${{ runner.temp }}/aptos-e2e-tests-devnet
+          command: cd ./crates/aptos/e2e && poetry run python main.py -d --base-network devnet --image-repo-with-project ${{ secrets.GCP_DOCKER_ARTIFACT_REPO }} --test-cli-tag ${{ inputs.GIT_SHA }} --working-directory ${{ runner.temp }}/aptos-e2e-tests-devnet
```

Comment:
you could maybe try a matrix? https://docs.github.com/en/actions/using-jobs/using-a-matrix-for-your-jobs

Reply:
also related, do you really need `runs-on: high-perf-docker` for these tests?

Reply:
So these do benefit from high-perf-docker because in total we pull down 4 different images as part of these tests, which is faster if we use the GCP artifact repo and we're inside GCP.

As for the matrix, running these in separate jobs would indeed be faster but it'd also be less efficient, since each job would have to pull down the image containing the CLI we're testing separately, whereas now the first one does it and then rest get to benefit from it being already pulled.